tests/std/tests/P0896R4_ranges_alg_count_if/test.cpp

```diff
@@ -13,41 +12,33 @@
 constexpr auto is_even = [](auto const& x) { return x % 2 == 0; };
 constexpr auto is_odd  = [](auto const& x) { return x % 2 != 0; };
 
-constexpr void smoke_test() {
-    using ranges::count_if;
-    using P                  = std::pair<int, int>;
-    std::array<P, 5> const x = {{{0, 47}, {1, 99}, {2, 99}, {3, 47}, {4, 99}}};
-    using D                  = ranges::range_difference_t<basic_borrowed_range<P const>>;
-
-    {
-        // Validate range overload
-        auto result = count_if(basic_borrowed_range{x}, is_even, get_first);
-        STATIC_ASSERT(std::same_as<decltype(result), D>);
-        assert(result == 3);
-    }
-    {
-        // Validate iterator + sentinel overload
-        basic_borrowed_range wrapped_x{x};
-        auto result = count_if(wrapped_x.begin(), wrapped_x.end(), is_odd, get_first);
-        STATIC_ASSERT(std::same_as<decltype(result), D>);
-        assert(result == 2);
-    }
-}
-
-int main() {
-    STATIC_ASSERT((smoke_test(), true));
-    smoke_test();
-}
+using namespace std;
+using P = pair<int, int>;
 
 struct instantiator {
-    template <class In>
-    static void call(In&& in = {}) {
-        using ranges::iterator_t;
-        using I = iterator_t<In>;
-
-        (void) ranges::count_if(in, UnaryPredicateFor<I>{});
-        (void) ranges::count_if(in, ProjectedUnaryPredicate<>{}, ProjectionFor<I>{});
+    static constexpr P input[5] = {{0, 47}, {1, 99}, {2, 99}, {3, 47}, {4, 99}};
+
+    template <class Read>
+    static constexpr void call() {
+        using ranges::count_if;
+        { // Validate iterator + sentinel overload
+            Read wrapped_input{input};
+
+            auto result = count_if(wrapped_input.begin(), wrapped_input.end(), is_odd, get_first);
+            STATIC_ASSERT(same_as<decltype(result), iter_difference_t<decltype(wrapped_input.begin())>>);

```

Comment:
I'd use `range_difference_t` here as on 35, since it's literally defined as `iter_difference_t<iterator_t<meow>>`.

Reply:
Done